File: domain/src/main/kotlin/domain/model/Invoice.kt
package domain.model

import java.time.LocalDateTime

enum class InvoiceStatus {
    PENDING,
    PAID,
    CANCELLED
}

data class Invoice(
    val id: InvoiceId,
    val operationId: OperationId,
    val amount: Money,
    val status: InvoiceStatus,
    val creationDateTime: LocalDateTime,
    val lastUpdateDateTime: LocalDateTime
)

@JvmInline
value class InvoiceId(val value: String) {
    override fun toString(): String = value
}

File: frontend/src/types/invoice.ts
import {Money} from "./Money";

export interface Invoice {
    id: string;
    operationId: string;
    amount: Money;
    status: InvoiceStatus;
    createdAt: string;
    updatedAt: string;
}

export enum InvoiceStatus {
    PENDING = 'PENDING',
    PAID = 'PAID',
    CANCELLED = 'CANCELLED',
}